### DISCUSSION TITLE
Select Dropdown List of Versions crashes the whole Visual Studio

### DISCUSSION THREAD

**Discussion by etg-al** [aws/aws-sdk-net#3131] [Q&A] [answered]
### Describe the issue

This package must have the worst versioning history ever! Almost one or 2 versions per day...It takes a lot of time just rendering the dropdown list of versions when updating from the Nuget Package Manager in Visual Studio! Sometimes it even crashes the whole IDE! I know there are workarounds about updating it via console, but It's hard for me to believe you really don't care at all about user experience.

### Links

![Capture](https://github.com/aws/aws-sdk-net/assets/96468047/fcab23da-7ac0-4d07-b40d-7e302f8c08fa)

**Comment by ashishdhingra** [comment]
@etg-al Good morning. Thanks for bringing this into our attention. This is the Visual Studio NuGet dialog issue which should, in essence support pagination or adaptive loading (or some other feature of such kind) to handle long list of version numbers. Could you try opening the issue in Microsoft's support forums to get the guidance? I'm unsure what we could do from our side, I'm unable to find any quick guidance on how Visual Studio NuGet dialog handles long lost in drop down or if other similar issue has been reported. However I would bring this to our team's attention.

We also noticed that the version dropdown is editable. As you type, it filters the drop down list:

![vs_nuget_filter](https://github.com/aws/aws-sdk-net/assets/67916761/3a9d111b-1f21-4bce-9d9f-6fdc1e1538bc)


Thanks,
Ashish

**Comment by dscpinheiro** [comment, marked as answer]
In addition to what Ashish said, please see this comment on a separate issue: https://github.com/aws/aws-sdk-net/issues/2554#issuecomment-1451042910

> We would like to get to that point of only releasing when the service changes. The reason it is happening right now is inside `AWSSDK.Core` there is an [embedded JSON file](https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Core/endpoints.json) that describes all of the AWS service endpoints. Given the number of AWS services we have these days it is almost daily there is some service region expansion causing that file to change. And when `AWSSDK.Core` changes we update all of the service packages to pick up the latest changes of `AWSSDK.Core`. 

> We are trying to move away the monolithic JSON file for several reasons but it takes time because there is a lot of automation in place used across many teams to create that file that gets released with almost every AWS SDK.

**Comment by etg-al** [comment]
@ashishdhingra thanks for your response. Your suggestion to address this issue to Microsoft sounds legit. It sounds legit in the same say it would if I code an algorithm with O(n!) complexity and blame the processor manufacturers why it takes so much time to complete. I mean, why do I have to try to code a more efficient algorithm, they can try to build faster chips..!
Yet somehow, common sense helps you believe that in fact I am doing something wrong, not them!

@dscpinheiro that's addressing the real issue, thanks for understanding my concern. I was sure these daily updates had nothing to do with actual code updates, and in a nutshell the issue I brought up is about a solution to decrease these absurd number of releases, as you say, by removing the embedded JSON form the build itself. It is a pain for us to tell which versions are service upgrades and which of them are just endpoint updates! I'm glad you have taken that into consideration and you are working on it!

Have a good day you both,
Andy.

**Comment by ashishdhingra** [comment]
Converting this to general discussion since we have a backlog to separate `endpoints.json`.
